### PR TITLE
Still implement StatisticsImpl::measureTime()

### PR DESCRIPTION
**Pull request by siying**
Summary:
Since Statistics::measureTime() is deprecated, StatisticsImpl::measureTime() is not implemented. We realized that users might have a wrapped Statistics implementation in which measureTime() is implemented as forwarded to StatisticsImpl, and causes assert failure. In order to make the change less intrusive, we implement StatisticsImpl::measureTime(). We will revisit whether we need to remove it after several releases.

Also, add a test to make sure that a Statistics implementation using the old interface still works.

Test Plan: Run all exiting tests.

### CODE REVIEW

File: monitoring/statistics.h [hunk 1]
```diff
@@ -53,6 +53,12 @@ class StatisticsImpl : public Statistics {
   virtual void setTickerCount(uint32_t ticker_type, uint64_t count) override;
   virtual uint64_t getAndResetTickerCount(uint32_t ticker_type) override;
   virtual void recordTick(uint32_t ticker_type, uint64_t count) override;
+  // Still get it implemented for now, in case a user explictly called it.
+  // For example, they may have a wrapped Statistics object, passing the call
+  // to recordTick() into here. We'll eventually remove the assert.
+  virtual void measureTime(uint32_t histogramType, uint64_t time) {
```
Comment (riversand963) [inline monitoring/statistics.h]: According to fbcode, we should favor `void measureTime(uint32_t histogramType, uint64_t time) override` from now on.

File: db/db_test2.cc [hunk 2]
```diff
@@ -3661,6 +3661,50 @@ TEST_F(DBTest2, MultiDBParallelOpenTest) {
 }
 #endif  // OS_WIN
 
+namespace {
+class DummyOldStats : public Statistics {
+ public:
+  uint64_t getTickerCount(uint32_t /*ticker_type*/) const override { return 0; }
+  void recordTick(uint32_t /* ticker_type*/, uint64_t /* count */) override {
```
Comment (riversand963) [inline db/db_test2.cc]: `/* ticker_type */`?

File: db/db_test2.cc [hunk 3]
```diff
@@ -3661,6 +3661,50 @@ TEST_F(DBTest2, MultiDBParallelOpenTest) {
...
+    num_rt++;
+  }
+  // Set is not supported for fb303 stats, we just forward to the stats object
```
Comment (riversand963) [inline db/db_test2.cc]: Are we going to test the forwarding logic here?

File: db/db_test2.cc [hunk 4]
```diff
@@ -3661,6 +3661,50 @@ TEST_F(DBTest2, MultiDBParallelOpenTest) {
...
+    return "";
+  }
+  bool HistEnabledForType(uint32_t /*type*/) const override { return false; }
+  std::string ToString() const override { return ""; }
+  int num_rt = 0;
+  int num_mt = 0;
+};
+}  // namespace
+
+TEST_F(DBTest2, OldStatsInterface) {
```
Comment (riversand963) [inline db/db_test2.cc]: In the test, isn't the way we use stats the same as before? Is the test name 'OldStatsInterface' a little ambiguous?

File: monitoring/statistics.h (hunk 1 above)
Comment (siying) [reply monitoring/statistics.h]: We should change fbcode's implementation of Statistics instead...

File: db/db_test2.cc (hunk 3 above)
Comment (siying) [reply db/db_test2.cc]: Thanks I will remove the comments.

File: db/db_test2.cc (hunk 4 above)
Comment (siying) [reply db/db_test2.cc]: Yes but I implemented a new one?

**Comment by siying** [comment]
Not sure what's the problem with Travis. It passed before I rebase. I ran clang and gcc tests locally and it is OK. I'm landing it to unblock 6.1 release.

**Pull request closed without merging**